dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -484,23 +484,24 @@ bool
 ModelMulti::propagateZModif() {
   vector<int> indicesDiff;
   silentZChange_ = false;
-  for (int i = 0; i < sizeZ(); ++i) {
-    bool isSilent = silentZ_[i];
-    if (isSilent && silentZChange_) {
-      continue;
-    } else if (isSilent) {
-      silentZChange_ =  doubleNotEquals(zLocal_[i], zSave_[i]);
-    } else if (doubleNotEquals(zLocal_[i], zSave_[i])) {
-      indicesDiff.push_back(i);
+  for (int i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
+    if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
+      indicesDiff.push_back(nonSilentZIndexes_[i]);
     }
   }
 
   bool changeDetected = !indicesDiff.empty();
   if (changeDetected) {
```

Comment:
Each time I see this part of code, I don't like the boolean changeDetected that is quite confusing (especially since we introduced the silentZ thing). 
I would suggest to directly test !indicesDiff.empty() and then to return true when necessary and false as a default value.

Reply:
Done